### PR TITLE
feat(cli): add host-aware `tap status` for first-step debugging

### PR DESCRIPTION
**Pull request by ggonzalez94**
## Summary
- New top-level `tap status` command that reports runtime state for a TAP data dir without starting any transport. Pure filesystem reads: transport owner lock, contacts, conversation logs, request journal, and the Hermes daemon state file.
- Separates handshake state (contacts) from message state (conversations) so "active contact with zero real messages" is immediately obvious — which was the exact confusion behind the recent Hermes debugging report.
- Detects host mode from the transport owner lock label (`hermes:*`, `openclaw:*`, `tap:listen`, other `tap:*`) and cross-references Hermes's configured identities, flagging the common multi-runtime pitfalls (stale lock, stuck `connecting` contact, queued commands with no owner, Hermes daemon running but not managing this data dir).

## Why
Today there is no single host-aware debug command. `tap hermes status` only covers Hermes, and everything else is spread across `tap contacts list`, `tap journal list`, `tap conversations list`, and the `.transport.lock` file. A reporter recently hit this while debugging a connection that succeeded at the handshake layer but showed no messages, and had no single place to confirm "did any real peer traffic happen vs just the handshake?". `tap status` answers that in one output.

## Key design calls
- **Filesystem-only.** No `createCliRuntime`, no transport start. The command has to work when the runtime is unhealthy — that's when you need it.
- **Config parsing is direct.** Uses `YAML.parse` against `config.yaml` rather than `loadTrustedAgentConfigFromDataDir`, because the loader defaults a missing `agent_id` to `0` which would misreport unregistered agents as `registered: true`. The command returns a discriminated `missing | invalid | parsed` state and has a dedicated `config-invalid` host mode with a distinct remediation ("fix or re-run `tap init`" instead of "run `tap register`").
- **Message counts come from the conversation logger.** `TapMessagingService.sendMessageInternal` writes to conversations but does NOT journal outbound `message/send` rows, so counting from the journal silently drops every successful outbound send. The conversation log is the source of truth.
- **Rejects `--config` outside `--data-dir`.** Same guard every other command runs via `loadConfig()` — prevents an internally-inconsistent status that reads contacts from agent A but reports `agent_id` from agent B.

## Test plan
- [x] 13 unit tests in `packages/cli/test/status.test.ts` covering: not-initialized, not-registered, idle, cli-listener, hermes-managed, Hermes misconfigured-for-this-data-dir, queued-commands-without-owner, handshake-vs-message separation, conversation-log-based counting, `config-invalid` from empty/garbage yaml, missing-agent_id-key, and `--config` path rejection.
- [x] Full workspace typecheck ✓
- [x] `bun run lint` ✓
- [x] Full CLI test suite (290/291 — one pre-existing `migrate-wallet` flake unrelated to these changes)
- [x] Smoke tested the built binary for text output, JSON output, `config-invalid` path, and cross-data-dir `--config` rejection

## Reviewer notes
- Does NOT update the `packages/cli/assets/skills/trusted-agents/SKILL.md` mirror even though `skills/trusted-agents/SKILL.md` changes. The mirror is already 134 lines out of sync with the canonical at `main`, so regenerating it in this PR would conflate the `tap status` docs update with unrelated pre-existing drift. The mirror catches up whenever someone runs `bun run --cwd packages/cli build`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Medium Risk**
> Moderate risk: introduces a new command that reads/parses multiple on-disk state files (locks, journals, conversation logs, Hermes state) and adds new warning/mode logic that could misreport state if edge cases are missed, though it is read-only and transport-free.
> 
> **Overview**
> Adds a new top-level `tap status` command that produces a host-aware, **filesystem-only** runtime snapshot for the selected TAP data dir (config validity/registration, transport lock owner + liveness, contact handshake state, conversation-based message counts, and request-journal pending/queued work).
> 
> Implements robust error-tolerant readers in `commands/status.ts` that *surface corruption as warnings* (e.g., unreadable `.transport.lock`, `contacts.json`, `request-journal.json`, or individual conversation files) and classifies host mode (`idle`, `cli-listener`, `hermes-managed`, `openclaw-managed`, `transport-unknown`, etc.), including Hermes identity/daemon cross-check warnings.
> 
> Adds a comprehensive `status.test.ts` suite covering mode detection, handshake-vs-message separation, queued-command/stale-lock scenarios, config edge cases (missing/empty/invalid YAML, missing `agent_id`), Hermes misconfiguration, and corruption handling, and updates `skills/trusted-agents/SKILL.md` to document `tap status` as the recommended first debug step.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit eeb245d3da3a232a66b020b9f6abef635fbcd7dd. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/cli/src/commands/status.ts [hunk 1]
```diff
@@ -0,0 +1,478 @@
...
+	} catch {
+		return [];
+	}
+}
+
+async function readJournal(dataDir: string): Promise<RequestJournalEntry[]> {
+	try {
+		return await new FileRequestJournal(dataDir).list();
+	} catch {
+		return [];
```
[the remark this answers is not in the document]
Comment (ggonzalez94) [reply packages/cli/src/commands/status.ts]: Agreed — fixed in e6e5412. `readJournal`/`readContacts`/`readConversations` now return a `{ value, error? }` shape and any error (parse / non-ENOENT I/O) is surfaced as an explicit warning ("Failed to read request-journal.json: … Journal counts are incomplete."). The underlying File* stores already treat ENOENT as empty, so anything bubbling up here is a real corruption the operator needs to see.

File: packages/cli/src/commands/status.ts [hunk 2]
```diff
@@ -0,0 +1,478 @@
...
+async function readHermesStatus(
+	hermesHomeOverride: string | undefined,
+	dataDir: string,
+): Promise<HermesStatus | null> {
+	const hermesHome = resolveHermesHome(hermesHomeOverride);
+	let config: TapHermesPluginConfig;
+	try {
+		config = await loadTapHermesPluginConfig(hermesHome);
+	} catch {
+		return null;
```
[the remark this answers is not in the document]
Comment (ggonzalez94) [reply packages/cli/src/commands/status.ts]: Agreed — fixed in e6e5412. `readHermesStatus` now distinguishes three states: (1) genuinely not installed → `null`, (2) config parse failure → returned with `config_error` populated and a dedicated warning ("Hermes TAP plugin config exists but cannot be parsed: …"), (3) normal installed. A daemon state file alone also keeps Hermes in the output so downstream warnings (daemon vs data-dir mismatch, etc.) still fire even if the config is broken.

File: packages/cli/src/commands/status.ts [hunk 3]
```diff
@@ -0,0 +1,478 @@
...
+		warnings.push(
+			`Contact ${input.contactsSummary.oldest_connecting.peer} has been in 'connecting' state for ${input.contactsSummary.oldest_connecting.age_minutes} minutes. Try \`tap message sync\`.`,
+		);
+	}
+
+	if (input.journalSummary.queued_commands > 0 && !input.transportOwner) {
+		warnings.push(
+			`${input.journalSummary.queued_commands} queued command(s) in journal but no transport owner is draining them. Start \`tap message listen\` or restart the host that owns this data dir.`,
+		);
+	}
```
[the remark this answers is not in the document]
Comment (ggonzalez94) [reply packages/cli/src/commands/status.ts:646]: Good catch — fixed in e6e5412. The condition is now `transportIdle = !owner || !owner.alive` so the queued-commands warning fires in both the no-lock and stale-lock cases. Added a test ("flags queued commands even when a stale (dead-pid) lock is present") asserting both warnings appear together.

File: packages/cli/src/commands/status.ts [hunk 4]
```diff
@@ -0,0 +1,563 @@
...
+	try {
+		return { value: await new FileRequestJournal(dataDir).list() };
+	} catch (err) {
+		return { value: [], error: formatReadError(err) };
+	}
+}
+
+async function readConversations(dataDir: string): Promise<ReadResult<ConversationLog[]>> {
+	try {
+		return { value: await new FileConversationLogger(dataDir).listConversations() };
```
[the remark this answers is not in the document]
Comment (ggonzalez94) [reply packages/cli/src/commands/status.ts]: Agreed — fixed in 74c88c8. `readConversations` no longer goes through `FileConversationLogger.listConversations` (which swallows per-file parse errors via its `// Skip corrupted files` branch). It now scans `conversations/` directly, reads each `.json` file, collects the failures, and surfaces them as "N conversation file(s) unreadable: conv-broken, ...". Added a test that seeds one good file + one garbage file and asserts the good file is still counted AND the bad file appears in warnings.

File: packages/cli/src/commands/status.ts [hunk 5]
```diff
@@ -0,0 +1,563 @@
...
+	hermesHomeOverride: string | undefined,
+	dataDir: string,
+): Promise<HermesStatus | null> {
+	const hermesHome = resolveHermesHome(hermesHomeOverride);
+
+	let daemonState: TapHermesDaemonState | null = null;
+	try {
+		daemonState = await readHermesTapDaemonState(hermesHome);
+	} catch {
+		daemonState = null;
```
[the remark this answers is not in the document]
Comment (ggonzalez94) [reply packages/cli/src/commands/status.ts]: Agreed — fixed in 74c88c8. `readHermesStatus` now captures `readHermesTapDaemonState` failures into a new `daemon_state_error` field on the Hermes status object (alongside the existing `config_error`), with its own warning ("Hermes TAP daemon state file exists but cannot be parsed: …. The daemon running state is unknown until this is resolved."). A broken daemon.json also keeps Hermes in the `installed` state so the warning path fires. Added a test that writes garbage into daemon.json and asserts the warning appears.

File: packages/cli/src/commands/status.ts [hunk 6]
```diff
@@ -0,0 +1,617 @@
...
+		// as "installed but broken" so the operator sees the root cause instead
+		// of a false "Hermes isn't installed" silence.
+		configError = formatReadError(err);
+	}
+
+	if (!config && !configError && !daemonState && !daemonStateError) {
+		// No config file, no broken config, no daemon state, no broken daemon
+		// state — genuinely not installed for this user.
+		return null;
+	}
```
[the remark this answers is not in the document]
Comment (ggonzalez94) [reply packages/cli/src/commands/status.ts]: Good catch — removed in f387a4e. You're right that `loadTapHermesPluginConfig` returns `{ identities: [] }` on ENOENT and throws otherwise, so `config === null` always implies `configError !== undefined`. Dropped the dead early-return and added a comment documenting the invariant so it doesn't get reintroduced. The genuine "not installed" path is the one you pointed at downstream.

File: packages/cli/src/commands/status.ts [hunk 7]
```diff
@@ -0,0 +1,617 @@
...
+		// Prevent an internally inconsistent status where --data-dir points at
+		// agent A but --config reads from agent B.
+		validateConfigPathInDataDir(opts, configPath, dataDir);
+
+		const configState = readConfigState(configPath);
+		const registered =
+			configState.kind === "parsed" && configState.agentId !== null && configState.agentId >= 0;
+
+		const ownerInspector = new TransportOwnerLock(dataDir, "tap:status");
+		const rawOwner = await ownerInspector.inspect();
```
[the remark this answers is not in the document]
Comment (ggonzalez94) [reply packages/cli/src/commands/status.ts]: Agreed — fixed in f387a4e. `TransportOwnerLock.inspect()` is now wrapped in a try/catch at the status-command level. A parse failure is captured as `transportOwnerError`, routed through the same `readErrors` pipe as the other file reads, and surfaced as a warning ("Failed to read .transport.lock: … Transport owner state is unknown.") while the command continues with `transport_owner: null`. Added a test that seeds a garbage `.transport.lock` and asserts the command still returns 200 with the warning.

File: packages/cli/src/commands/status.ts [hunk 8]
```diff
@@ -0,0 +1,617 @@
...
+	}
+
+	const value: ConversationLog[] = [];
+	const failed: string[] = [];
+	for (const entry of entries) {
+		if (!entry.endsWith(".json")) continue;
+		const filePath = join(conversationsDir, entry);
+		try {
+			const raw = await readFile(filePath, "utf-8");
+			value.push(JSON.parse(raw) as ConversationLog);
```
[the remark this answers is not in the document]
Comment (ggonzalez94) [reply packages/cli/src/commands/status.ts]: Agreed — fixed in f387a4e. Added an `isConversationLog` type guard that checks `messages` is an array plus `peerAgentId`/`peerDisplayName` types. Schema-invalid files (`{}`, `{"messages": null}`, etc.) are now treated like parse failures and counted in the "N conversation file(s) unreadable" warning. Added a test seeding both `{}` and `{"messages": null, ...}` that asserts the command returns 200 with the warning rather than crashing `summarizeMessages`.

File: packages/cli/src/commands/status.ts [hunk 9]
```diff
@@ -0,0 +1,649 @@
...
+
+function isConversationLog(value: unknown): value is ConversationLog {
+	// Minimal shape check — just enough so `summarizeMessages` can iterate
+	// safely. A file like `{}` or `{"messages": null}` is schema-invalid and
+	// must be counted as "unreadable" rather than crashing the whole command.
+	if (typeof value !== "object" || value === null) return false;
+	const record = value as Record<string, unknown>;
+	if (!Array.isArray(record.messages)) return false;
+	if (typeof record.peerAgentId !== "number") return false;
+	if (typeof record.peerDisplayName !== "string") return false;
```
[the remark this answers is not in the document]
Comment (ggonzalez94) [reply packages/cli/src/commands/status.ts:354]: Agreed — fixed in 2c67f2b. `isConversationLog` now validates each entry in `messages`: must be an object, must have `direction` equal to `incoming` or `outgoing`, and must have a string `timestamp`. Files like `{"messages": [null]}` or one where a message is missing `direction` are now counted in the "N conversation file(s) unreadable" warning instead of crashing `summarizeMessages`. Extended the test to seed `conv-null-element.json` and `conv-bad-direction.json`.

File: packages/cli/src/commands/status.ts [hunk 10]
```diff
@@ -0,0 +1,649 @@
...
+	}
+
+	// Queued commands are equally undrained whether no lock exists OR the lock
+	// belongs to a dead process. A truthy-but-`alive: false` owner must NOT
+	// swallow this warning — the operator needs to see both the stale-lock
+	// note and the "nothing is draining your queue" note.
+	const transportIdle = !input.transportOwner || !input.transportOwner.alive;
+	if (input.journalSummary.queued_commands > 0 && transportIdle) {
+		warnings.push(
+			`${input.journalSummary.queued_commands} queued command(s) in journal but no live transport owner is draining them. Start \`tap message listen\` or restart the host that owns this data dir.`,
```
[the remark this answers is not in the document]
Comment (ggonzalez94) [reply packages/cli/src/commands/status.ts:638]: Agreed — fixed in 2c67f2b. The queued-commands warning now requires `transportOwnershipKnown` (i.e. `readErrors.transportOwner === undefined`) in addition to `transportIdle`. When lock inspection fails we only emit the existing "Failed to read .transport.lock" warning and suppress the "nothing is draining your queue" diagnosis, since a live host could still be draining it — we just can't see. Added a test that seeds a garbage lock + a queued command and asserts the lock warning fires but the queued-commands warning does not.

File: packages/cli/src/commands/status.ts [hunk 11]
```diff
@@ -0,0 +1,665 @@
...
+
+	if (input.hermes?.daemon_state_error) {
+		warnings.push(
+			`Hermes TAP daemon state file exists but cannot be parsed: ${input.hermes.daemon_state_error}. The daemon running state is unknown until this is resolved.`,
+		);
+	}
+
+	if (input.hermes?.daemon_running && !input.hermes.manages_this_data_dir) {
+		warnings.push(
+			"Hermes TAP daemon is running but this data-dir is not in its configured identities. If you expected Hermes to own this agent's transport, run `tap hermes configure --name <name>` from this data dir.",
```
[the remark this answers is not in the document]
Comment (ggonzalez94) [reply packages/cli/src/commands/status.ts]: Agreed — fixed in 0773b77. The "daemon is running but this data-dir is not in configured identities" warning is now gated on `hermesConfigReadable` (`config_error === undefined`). When config.json is unreadable the existing config-error warning already tells the operator what to fix; we no longer pile on a false "missing identity" diagnosis because we couldn't actually read the identity list. Added a test that seeds a garbage config.json + a live daemon state and asserts the identity-mismatch warning does NOT fire.

File: packages/cli/src/commands/status.ts [hunk 12]
```diff
@@ -0,0 +1,665 @@
...
+		);
+	}
+
+	if (input.hermes?.manages_this_data_dir && !input.hermes.daemon_running) {
+		warnings.push(
+			"Hermes is configured to manage this data-dir but the TAP daemon is not running. Start or restart `hermes gateway`.",
```
[the remark this answers is not in the document]
Comment (ggonzalez94) [reply packages/cli/src/commands/status.ts]: Agreed — fixed in 0773b77. Same pattern — the "daemon is not running" warning is now gated on `hermesDaemonStateReadable` (`daemon_state_error === undefined`). When daemon.json is unreadable we only emit the existing "running state is unknown until this is resolved" warning and suppress the contradictory "daemon is not running" diagnosis. Added a test that seeds a valid plugin config + garbage daemon.json and asserts the stop/restart warning does NOT fire.

File: packages/cli/src/commands/status.ts [hunk 13]
```diff
@@ -0,0 +1,665 @@
...
+		);
+	}
```
[the remark this answers is not in the document]
Comment (ggonzalez94) [reply packages/cli/src/commands/status.ts:679]: Agreed — fixed in 0773b77. Same underlying issue as the Codex comment right next to this one. The "daemon is not running" warning now has the same `hermesDaemonStateReadable` guard the transport-lock path uses via `transportOwnershipKnown`. When `daemon.json` is corrupt, we only emit the "running state is unknown" warning and suppress the contradictory stop/restart diagnosis.

File: packages/cli/src/commands/status.ts [hunk 14]
```diff
@@ -0,0 +1,680 @@
...
+	configState: ConfigState;
+	registered: boolean;
+	transportOwner: (TransportOwnerInfo & { alive: boolean }) | null;
+}): HostMode {
+	if (input.configState.kind === "missing") return "not-initialized";
+	if (input.configState.kind === "invalid") return "config-invalid";
+	if (!input.registered) return "not-registered";
+
+	const owner = input.transportOwner;
+	if (!owner || !owner.alive) return "idle";
```
[the remark this answers is not in the document]
Comment (ggonzalez94) [reply packages/cli/src/commands/status.ts:451]: Agreed — fixed in eeb245d. Added a new `transport-unknown` entry to the `HostMode` union, and `detectMode` now returns it (before the regular owner-label dispatch) whenever `transportOwnerError` is set. JSON consumers that key off `host.mode` alone will no longer be told "idle" when ownership is explicitly unknown. Extended the existing corrupt-lock test to assert the new mode.

File: packages/cli/test/status.test.ts [hunk 15]
```diff
@@ -0,0 +1,762 @@
...
+		hermesHome = join(tempRoot, "hermes-home");
+		await mkdir(hermesHome, { recursive: true });
+		originalHermesHome = process.env.HERMES_HOME;
+		process.env.HERMES_HOME = hermesHome;
+		process.exitCode = undefined;
+	});
+
+	afterEach(async () => {
+		if (originalHermesHome === undefined) {
+			process.env.HERMES_HOME = undefined;
```
[the remark this answers is not in the document]
Comment (ggonzalez94) [reply packages/cli/test/status.test.ts]: Good catch — fixed in eeb245d. Switched the afterEach to `delete process.env.HERMES_HOME` with a targeted `// biome-ignore lint/performance/noDelete` because biome's `noDelete` rule would otherwise push the bug back in. The prior `= undefined` was leaking the string "undefined" into later suites.
